app/controllers/hosts_controller.rb

```diff
@@ -273,19 +267,16 @@ def power
   end
 
   def get_power_state
-    result = {:id => @host.id}.merge(host_power_state(:na))
-    if @host.supports_power?
-      result = host_power_ping result
-    else
-      result[:statusText] = _('Power operations are not enabled on this host.')
-    end
-
-    render :json => result
+    render json: PowerManager::PowerStatus.new(host: @host).power_state
```

Comment:
I think we can probably remove the whole code from the hosts controller, and change the react code to simply consume the api instead?

Reply:
Ideally, the react code would consume the graphql API. ;-) What I'm trying to say is: Let's do this in a separate issue, it's out of scope of this PR.

Reply:
while I don't object, I don't see value in having two controllers exposing this info, if the api result is identical to the ui controller, it should be as easy as changing the url the react prop is receiving? (https://github.com/theforeman/foreman/blob/develop/app/views/hosts/_list.html.erb#L30) (anything we can do to reduce the size of the host controller/model is welcomed :)

Reply:
@leewaa: I'd say that's your call. If you want to change the route in the react code: Feel free to do so. Nobody's gonna stop ya!

Reply:
:D ok, I will take a look. If its just a simple change then I ill go ahead with it and just change it here.